api/src/shop/stripe_product_price.py

```diff
@@ -68,34 +76,50 @@ def get_stripe_prices(
 
 def eq_makeradmin_stripe_product(makeradmin_product: Product, stripe_product: stripe.Product) -> bool:
     """Check that the essential parts of the product are the same in both makeradmin and stripe"""
+    if stripe_product.id != str(makeradmin_product.stripe_product_id):
+        raise ValueError(
+            f"Stripe product id {stripe_product.id} and makeradmin product id {makeradmin_product.id} does not match"
+        )
     return stripe_product.name == makeradmin_product.name
 
 
 def eq_makeradmin_stripe_price(makeradmin_product: Product, stripe_price: stripe.Price, price_type: PriceType) -> bool:
     """Check that the essential parts of the price are the same in both makeradmin and stripe"""
+    lookup_key_for_product = get_stripe_price_lookup_key(makeradmin_product, price_type)
+    if stripe_price.lookup_key != lookup_key_for_product:
+        raise ValueError(
+            f"Stripe price lookup key {stripe_price.lookup_key} and corresponding key from"
+            + f" makeradmin product {makeradmin_product.id} and PriceType {price_type} does not match"
+        )
+
     recurring = makeradmin_to_stripe_recurring(makeradmin_product, price_type)
     different = []
 
-    if recurring:
+    if recurring is not None:
         if stripe_price.recurring is None:
             return False
         different.append(stripe_price.recurring.get("interval") != recurring.interval)
         different.append(stripe_price.recurring.get("interval_count") != recurring.interval_count)
+    else:
+        if stripe_price.recurring is not None:
+            return False
     different.append(stripe_price.unit_amount != stripe_amount_from_makeradmin_product(makeradmin_product, recurring))
     different.append(stripe_price.currency != CURRENCY)
     different.append(stripe_price.metadata.get("price_type") != price_type.value)
```

Comment:
```suggestion
    elif stripe_price.recurring is None:
        different.append(stripe_price.unit_amount != stripe_amount_from_makeradmin_product(makeradmin_product, recurring))
        different.append(stripe_price.currency != CURRENCY)
        different.append(stripe_price.metadata.get("price_type") != price_type.value)
```

Why do we do this check? Seems unintuitive.